src/com/twitter/intellij/pants/service/project/model/ProjectInfo.java

```diff
@@ -180,4 +180,12 @@ public String toString() {
            ", targets=" + targets +
            '}';
   }
+
+  public void merge(ProjectInfo projectInfo) {
+    getLibraries().putAll(projectInfo.getLibraries());
+    getTargets().putAll(projectInfo.getTargets());
```

Comment:
I'd prefer to use the instance variables directly get as an indicator it is modifying something within the class.

`getLibraries()` and `getTargets()` really mean to return a frozen collection. Sorry legacy issue again.